docs/x509.rst

```diff
@@ -798,6 +800,97 @@ X.509 Extensions
 
         Where to access the information defined by the access method.
 
+.. class:: CRLDistributionPoints
+
+    .. versionadded:: 0.9
+
+    The CRL distribution points extension identifies how CRL information is
+    obtained. It is an iterable, containing one or more
+    :class:`DistributionPoint` instances.
+
+.. class:: DistributionPoint
+
+    .. versionadded:: 0.9
+
+    .. attribute:: full_name
+
+        :type: list of :class:`GeneralName` instances or None
+
+        This field describes methods to retrieve the CRL. If this is not None
+        then ``relative_name`` must be None.
+
+    .. attribute:: relative_name
+
+        :type: :class:`Name` or None
+
+        This field describes methods to retrieve the CRL relative to the CRL
+        issuer. If this is not None then ``full_name`` must be None.
```

Comment:
"At most one of ... or ... will be non-None"?

On Sat, May 9, 2015 at 1:12 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1906#discussion_r29994448>:
>
> > +
> > +    .. versionadded:: 0.9
> > +
> > +    .. attribute:: full_name
> > +
> > +        :type: list of :class:`GeneralName` instances or None
> > +
> > +        This field describes methods to retrieve the CRL. If this is not None
> > +        then ``relative_name`` must be None.
> > +
> > +    .. attribute:: relative_name
> > +
> > +        :type: :class:`Name` or None
> > +
> > +        This field describes methods to retrieve the CRL relative to the CRL
> > +        issuer. If this is not None then ``full_name`` must be None.
>
> Both can be none though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1906/files#r29994448>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084